### DISCUSSION TITLE
How can I prevent GitHub copilot from using my code

### DISCUSSION THREAD

**Discussion by leofleeo** [community/community#39361] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I would not like GitHub copilot to use my code for code snippets. Is there a way to prevent my code from being used by GitHub copilot?

**Comment by yongjulejule** [comment, marked as answer]
in `settings` -> `copilot`, you can change option

**Reply by leofleeo** [reply]
It says I need to buy GitHub copilot to change the setting

**Reply by leofleeo** [reply]
![Screenshot_2022-11-18-07-12-09-17_3aea4af51f236e4932235fdada7d1643.jpg](https://user-images.githubusercontent.com/83542885/202702542-b31b0d5f-171d-4cb1-9a4c-ae90c59353af.jpg)

**Reply by leofleeo** [reply]
Doesn't give me access to the setting >:(

**Reply by leofleeo** [reply]
Also, I mean the code in my GitHub repos

**Reply by leofleeo** [reply]
I just don't want people having to pay to find my code

**Comment by MulleDK19** [comment]
Only way is to make your repo private.

**Comment by cmp-nct** [comment]
If you don't want it to use your code then do not host it on Microsoft servers. Use a private GIT repository on an own server or vhost (it's free on GCE or AWS for example).
If you want to host a public visible project then you can not stop Copilot from reading it. Machine learning is similar to human learning in that way: we look at code and learn from it. 
If you make it public, you also agree that people look at it.

**Reply by leofleeo** [reply]
Switching to codeberg :)

**Reply by cmp-nct** [reply]
That makes no difference if you open source your project, it will still be used for copilot/codex , bloom and similar networks to train.
